### PR TITLE
Update CA1010 to have more global message to match new user option

### LINKED ISSUE #3274: Rename CollectionsShouldImplementGenericInterface.cs
Follow-up action from PR #3258

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #3274 

Docs will need to be updated.

### CODE REVIEW

**Comment by Evangelink** [comment]
Tagging @sharwell @mavasani for review

**Comment by Evangelink** [comment]
Ping @sharwell

File: src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CSharpGenericInterfaceShouldAlsoBeImplemented.Fixer.cs [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
 namespace Microsoft.CodeQuality.CSharp.Analyzers.ApiDesignGuidelines
 {
     /// <summary>
-    /// CA1010: Collections should implement generic interface
+    /// CA1010: Generic interfaces should be implemented
     /// </summary>
     [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
-    public sealed class CSharpCollectionsShouldImplementGenericInterfaceFixer : CollectionsShouldImplementGenericInterfaceFixer
+    public sealed class CSharpGenericInterfacesShouldAlsoBeImplementedFixer : GenericInterfacesShouldAlsoBeImplementedFixer
```
Comment (mavasani) [inline src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CSharpGenericInterfaceShouldAlsoBeImplemented.Fixer.cs]: The existing type names were chosen to exactly match the type names in legacy implementation so it acts as an easy reference point. Additionally, the docs also use the prior name: https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1010?view=vs-2019.

**Comment by mavasani** [comment]
@Evangelink There seem to be merge conflicts that need to be resolved.

**Comment by stephentoub** [comment]
I didn't merge this.  I don't know what happened.  I merged https://github.com/dotnet/roslyn-analyzers/pull/3001, and GitHub appears to be having issues and merged this one as well on my behalf!

**Comment by mavasani** [comment]
@stephentoub No worries. I was trying to merge this since morning, but GitHub was acting up on me. Likely your other merge triggered something to cause this one to get merged :-)

**Comment by stephentoub** [comment]
Yeah, GitHub's been having issues all morning; https://www.githubstatus.com/ is still reporting it on-going.

**Pull request merged**